build-scripts/build_hs_lf_tooling.sh

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env bash
+# Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set -eou pipefail
+
+if [ "$#" -ne 1 ]; then
+    echo "Expected exactly one argument."
+    echo "Usage: ./build_packages.sh TARGET_DIR"
+    exit 1
+fi
+
+TARGET_DIR=$PWD/$1
+cd "$(dirname ${BASH_SOURCE[0]})/.."
+
+mkdir -p $TARGET_DIR
+
+package_from_dir() {
+    local dir=$1
+    if [ ! -d $dir ]; then
+        echo "Directory $dir does not exist!"
+        exit 1
+    fi
+    pushd $dir
+    cabal new-sdist
+    cp dist-newstyle/sdist/*.tar.gz "$TARGET_DIR/"
+    rm -rf cabal.tix dist-newstyle/
+    popd
+}
+
+package_from_dir libs-haskell/da-hs-base
+# removed the GCP logger from the exposed modules to avoid SdkVersion.hs
+
+package_from_dir compiler/daml-lf-ast
+
+package_from_dir compiler/daml-lf-proto
+
+package_from_dir compiler/daml-lf-reader
+
+DIR=$(mktemp -d)
+mkdir -p "$DIR/protobuf/com/daml"
+cp -RL "daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev" "$DIR/protobuf/com/daml/"
+# generate code from protobuf using the matching tool, as a configure script
+cat <<EOF >"$DIR/Setup.hs"
```

Comment:
Custom setup scripts always end up causing problems at some point ime. Why did you decide against the approach the old script used which just used our Bazel setup to build the generated code and included that?

Reply:
The extracted library uses `proto-suite-0.4.0.0` instead of the point between-versions used in the daml repo.
To use the released version, I need the packaged protobuf compiler => calling it in a custom setup.

Reply:
Can we just switch t a released version in the DAML repo? The more you diverge from the build setup used in the DAML Repo the more likely this is to break in the future.

Reply:
Happy to assist you in switching. I don't want to make that change on this PR, though.
IIUC it should be fine to just update the bazel configuration to depend on the (hackage-released) `proto-suite-0.4.0.0`. Is that right @garyverhaegen-da ? I can confirm all code in `daml-lf-proto` written against the generated  `Com.Daml.DamlLfDev.DamlLf*` works with this version.

Reply:
I'm actually not very familiar with that side of things, but it seems to me we _are_ depending on the release 0.4.0.0 version of `proto3-suite` per [stack-snapshot.yml](https://github.com/digital-asset/daml/blob/master/stack-snapshot.yaml#L26). What am I missing? Where is that "in-between version" we're using defined?